src/Asset.js

```diff
@@ -50,7 +50,7 @@ class Asset {
   async getDependencies() {
     await this.loadIfNeeded();
 
-    if (this.mightHaveDependencies()) {
+    if (this.mightHaveDependencies() && this.contents) {
```

Comment:
`mightHaveDependencies` already checks for `this.contents`, so this additional check is not required.

Reply:
Some Asset class like CSSAsset override `getDependencies` method or some others may override `getDependencies`, this is wanna double check if contents needs to be parsed, trying not to modifiy other Asset class.

Reply:
swap the order of these conditions so `mightHaveDependencies` doesn't run if `this.contents` is empty.